GG/src/GUI.cpp

```diff
@@ -316,28 +316,28 @@ void GUIImpl::HandleMouseButtonPress(unsigned int mouse_button, const Pt& pos, i
     m_mouse_button_state[mouse_button] = true;
     m_drag_wnds[mouse_button] = m_curr_wnd_under_cursor; // track this window as the one being dragged by this mouse button
     if (m_curr_wnd_under_cursor) {
-        m_prev_wnd_drag_position = m_drag_wnds[mouse_button]->UpperLeft();
+        m_prev_wnd_drag_position = m_curr_wnd_under_cursor->UpperLeft();
```

Comment:
It is not a change in logic. Line 317 is `m_drag_wnds[mouse_button] = m_curr_wnd_under_cursor;`, so they point to the same window.  `m_curr_wnd_under_cursor` is set from `CheckedGetWindowUnder` at the beginning of this function.  It seemed clearer to continue referring to it by its first name.  

It anticipated the bug fix for https://github.com/freeorion/freeorion/issues/1639.  THe commit, https://github.com/freeorion/freeorion/pull/1632/commits/d8f60387906bacd1e8184c65e2db40b53b16d4d8, changes the code so that an additional mouse button click while a window is already being dragged does not start a new drag.  `curr_wnd_under_cursor` and  are not the same `m_drag_wnds[mouse_button]`.